File: app/src/main/java/com/otoniel/alura/dao/StudentDAO.java
package com.otoniel.alura.dao;

import com.otoniel.alura.model.Student;

import java.util.ArrayList;
import java.util.List;

public class StudentDAO {

    private final static List<Student> listStudents = new ArrayList<>();

    private int getNextStudentId() {
        return listStudents.size() + 1;
    }

    public void save(Student student) {
        listStudents.add(new Student(
                getNextStudentId(),
                student.getNome(),
                student.getPhone(),
                student.getEmail()
        ));
    }

    public void edit(Student student) {
        int indexStudent = getIndexStudentById(student.getId());
        if (indexStudent != -1) {
            listStudents.set(indexStudent, student);
        }
    }

    public void delete(Student student) {
        final int index = getIndexStudentById(student.getId());
        if (index != -1) {
            final Student studentFromList = listStudents.get(index);
            listStudents.remove(studentFromList);
        }
    }

    private int getIndexStudentById(int studentId) {
        int studentIndex = -1;
        for (Student mStudent : listStudents) {
            if (studentId == mStudent.getId()) {
                studentIndex = listStudents.indexOf(mStudent);
                break;
            }
        }
        return studentIndex;
    }

    public List<Student> getAll() {
        return new ArrayList<>(listStudents);
    }
}

File: app/src/main/java/com/otoniel/alura/dao/StudentDAO.kt
package com.otoniel.alura.dao

import com.otoniel.alura.model.Student
import java.util.ArrayList

class StudentDAO {
    private val nextStudentId: Int
        get() = listStudents.size + 1

    fun save(student: Student) {
        listStudents.add(Student(
            nextStudentId,
            student.nome,
            student.phone,
            student.email
        ))
    }

    fun edit(student: Student) {
        val indexStudent = getIndexStudentById(student.id)
        if (indexStudent != -1) {
            listStudents[indexStudent] = student
        }
    }

    fun delete(student: Student) {
        val index = getIndexStudentById(student.id)
        if (index != -1) {
            val studentFromList = listStudents[index]
            listStudents.remove(studentFromList)
        }
    }

    private fun getIndexStudentById(studentId: Int): Int {
        var studentIndex = -1
        for (mStudent in listStudents) {
            if (studentId == mStudent.id) {
                studentIndex = listStudents.indexOf(mStudent)
                break
            }
        }
        return studentIndex
    }

    val all: List<Student>
        get() = ArrayList(listStudents)

    companion object {
        private val listStudents: MutableList<Student> = ArrayList()
    }
}